packages/video_player/video_player_web/test/video_player_web_test.dart

```diff
@@ -82,6 +83,24 @@ void main() {
       expect(VideoPlayerPlatform.instance.play(textureId), completes);
     });
 
+    test('throws PlatformException when playing bad media', () async {
+      int videoPlayerId = await VideoPlayerPlatform.instance.create(
+        DataSource(
+            sourceType: DataSourceType.network,
+            uri:
+                'https://flutter.github.io/assets-for-api-docs/assets/videos/_non_existent_video.mp4'),
+      );
+
+      Stream<VideoEvent> eventStream =
+          VideoPlayerPlatform.instance.videoEventsFor(videoPlayerId);
+
+      // Mute video to allow autoplay (See https://goo.gl/xX8pDD)
+      await VideoPlayerPlatform.instance.setVolume(videoPlayerId, 0);
+      await VideoPlayerPlatform.instance.play(videoPlayerId);
+
+      expect(eventStream.toList(), throwsA(isA<PlatformException>()));
```

Comment:
consider using [`emitsError`](https://pub.dev/documentation/test_api/latest/test_api/emitsError.html) here

Reply:
Ohh I didn't know that matcher. Good point. Updated, and waiting for the tests to go green.